app/upgrades/v3_0/upgrades.go

```diff
@@ -0,0 +1,178 @@
+//nolint:revive,stylecheck
+package v3_0
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"slices"
+
+	sdkmath "cosmossdk.io/math"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/mezo-org/mezod/app/upgrades"
+	"github.com/mezo-org/mezod/utils"
+	evmkeeper "github.com/mezo-org/mezod/x/evm/keeper"
+	evmtypes "github.com/mezo-org/mezod/x/evm/types"
+	feemarketkeeper "github.com/mezo-org/mezod/x/feemarket/keeper"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	keepers *upgrades.Keepers,
+) upgradetypes.UpgradeHandler {
+	return func(
+		ctx context.Context,
+		_ upgradetypes.Plan,
+		fromVM module.VersionMap,
+	) (module.VersionMap, error) {
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+		sdkCtx.Logger().Info("running v3.0.0 upgrade handler")
+
+		err := updateMinGasPrice(sdkCtx, keepers.FeeMarketKeeper)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update minimum gas price: %w", err)
+		}
+
+		err = updateMaintenancePrecompileVersion(sdkCtx, keepers.EvmKeeper)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update maintenance precompile version: %w", err)
+		}
+
+		err = updateBridgePrecompileVersion(sdkCtx, keepers.EvmKeeper)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update bridge precompile version: %w", err)
+		}
+
+		return mm.RunMigrations(ctx, configurator, fromVM)
+	}
+}
+
+func updateMinGasPrice(sdkCtx sdk.Context, feeMarketKeeper feemarketkeeper.Keeper) error {
+	chainID := sdkCtx.ChainID()
+	params := feeMarketKeeper.GetParams(sdkCtx)
+
+	sdkCtx.Logger().Info(
+		"begin minimum gas price update",
+		"chainID", chainID,
+		"minGasPrice", params.MinGasPrice,
+	)
+
+	var newMinGasPrice sdkmath.LegacyDec
+	var err error
+
+	//nolint:gocritic
+	if utils.IsMainnet(chainID) {
+		newMinGasPrice, err = sdkmath.LegacyNewDecFromStr("1300000.000000000000000000")
+		if err != nil {
+			return err
+		}
+	} else if utils.IsTestnet(chainID) {
+		newMinGasPrice, err = sdkmath.LegacyNewDecFromStr("130.000000000000000000")
```

Comment:
Would there be any reason to align mainnet and testnet parameters here? It'd probably make things a bit more predictible for people developing contracts and testing them on testnet regarding gas costs and such.